pkg/provider/vault/auth_iam.go

```diff
@@ -183,3 +142,75 @@ func (c *client) requestTokenWithIamAuth(ctx context.Context, iamAuth *esv1.Vaul
 	}
 	return nil
 }
+
+func (c *client) getRegionOrDefault(region string) string {
+	if region != "" {
+		return region
+	}
+	return defaultAWSRegion
+}
+
+func (c *client) getAuthMountPathOrDefault(path string) string {
+	if path != "" {
+		return path
+	}
+	return defaultAWSAuthMountPath
+}
+
+func (c *client) getControllerPodCredentials(ctx context.Context, region string, k kclient.Client, jwtProvider util.JwtProviderFactory) (*credentials.Credentials, error) {
+	// First try IRSA (Web Identity Token) - checking if controller pod's service account is IRSA enabled
+	tokenFile := os.Getenv(vaultiamauth.AWSWebIdentityTokenFileEnvVar)
+	if tokenFile != "" {
+		logger.V(1).Info("using IRSA token for authentication")
+		return c.getCredsFromIRSAToken(ctx, tokenFile, region, k, jwtProvider)
+	}
+
+	// Check for Pod Identity environment variables.
+	podIdentityURI := os.Getenv(vaultiamauth.AWSContainerCredentialsFullURIEnvVar)
+	podIdentityToken := os.Getenv(vaultiamauth.AWSContainerAuthorizationTokenFileEnvVar)
+
+	if podIdentityURI != "" && podIdentityToken != "" {
+		logger.V(1).Info("using Pod Identity for authentication")
+		// Return nil to let AWS SDK v1 container credential provider handle Pod Identity automatically
+		return nil, nil
+	}
+
+	// No IRSA or Pod Identity found.
+	return nil, fmt.Errorf(errNoAWSAuthMethodFound)
```

Comment:
I'm not sure why this wasnt flagged by CI - but since we're not `Sprintf`ing, this would return an error
```suggestion
	return nil, errors.New(errNoAWSAuthMethodFound)
```

Reply:
not caught by CI because it doesn't actually error because there are no format verbs in that string, but absoluteely agree it can justt be an errors.New()